tools/cdk-release/lib/conventional-commits.ts

```diff
@@ -60,7 +60,15 @@ export interface ConventionalCommit {
  * @param gitTag the string representing the Git tag,
  *   will be used to limit the returned commits to only those added after that tag
  */
-export async function getConventionalCommitsFromGitHistory(gitTag: string): Promise<ConventionalCommit[]> {
+export async function getConventionalCommitsFromGitHistory(args: ReleaseOptions, gitTag: string): Promise<ConventionalCommit[]> {
+  // Since the commits are needed mainly for the Changelog generation,
+  // skip getting them if skipChangelog is `true`.
+  // This is needed to make our build succeed in environments without a Git repository,
+  // like CodeBuild in CodePipeline
+  if (args.skip?.changelog) {
```

Comment:
Adding this here feels a bit impure, is there an argument for moving this function call to internal to the changelog function instead of in the main function?

Reply:
That's where it was originally, but it was changed after https://github.com/aws/aws-cdk/pull/15331#discussion_r660396909 😜

Reply:
then maybe we hoist the conditional to the main method?
```
bump()

if (!args.skip?.changelog) {
  getCommits()
  filterCommits()
  changelog()
}

commit()
```